Mechty-learn-frontend/src/Components/threejs-components/Rocket.tsx

```diff
@@ -0,0 +1,97 @@
+import * as THREE from 'three'
+import { GLTF, GLTFLoader } from 'three/addons/loaders/GLTFLoader.js'
+import { useEffect, useRef } from "react"
+
+function Rocket() {
+    const refContainer = useRef<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+        if (!refContainer.current) return
```

Comment:
The code inside the callback passed to the `useEffect`  seems like a really long one. I suggest you break up this big functions into smaller ones. If you'll have multiple components using THREE.js you'll sure to find reusable functions.